Q/qdldl/build_tarballs.jl

```diff
@@ -0,0 +1,45 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "qdldl"
+version = v"0.1.5"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/osqp/qdldl/archive/refs/tags/v$(version).tar.gz", "2868b0e61b7424174e9adef3cb87478329f8ab2075211ef28fe477f29e0e5c99")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd qdldl-0.1.5/
+mkdir build_double
+cd build_double/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix/double -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DDFLOAT=0 ..
+cmake --build . -j8
+make install
+cd ..
+mkdir build_single
+cd build_single/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix/single -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DDFLOAT=1 ..
+cmake --build . -j8
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libqdldl", :libqdldl64, "double/lib"),
+    LibraryProduct("libqdldl", :libqdldl32, "single/lib")
```

Comment:
Unfortunately it isn't a good idea to have two libraries with same soname in the same process.  It's hugely problematic: #3801

Reply:
@imciner2 you seem to be familiar with this library.  Is there a way to build the two distinct variants next to each other with different sonames by default?  This is what FFTW does, which makes it possible for different variants to coexist, but having same soname only causes problems.

Reply:
Bummer. I'll see if I can get the authors to provide an option to add a suffix to the library name.

Reply:
The short answer is no, there is no way to build the two variants next to each other right now, and I don't think any other consumer of QDLDL has ever wanted to provide both a float32 and float64 variant, they usually have just used float64 if they provide a library (or at least no one has reached out to us about it). Is there a real need to provide float32 in the C library when there is QDLDL.jl that can handle any type under `AbstractFloat` since it is a pure Julia implementation?